project/geosource/models.py

```diff
@@ -76,13 +123,16 @@ class Source(PolymorphicModel, CeleryCallMethodsMixin):
 
     settings = models.JSONField(default=dict, blank=True)
     groups = models.ManyToManyField(Group, blank=True, related_name="geosources")
-    report = models.JSONField(default=dict, blank=True, encoder=DjangoJSONEncoder)
+    report = models.OneToOneField(SourceReporting, on_delete=models.SET_NULL, null=True)
 
     task_id = models.CharField(null=True, max_length=255, blank=True)
     task_date = models.DateTimeField(null=True, blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
     last_refresh = models.DateTimeField(default=timezone.now)
+    status = models.PositiveIntegerField(
+        choices=Status.choices, null=True, max_length=20
```

Comment:
```suggestion
        choices=Status.choices, null=True
```
no need to get a max_length on an integer. Bu you can use PositiveSmallIntegerField

Reply:
Oh, that's nice. I will definitively use `PositiveSmallIntegerField`